js/utilities.js

```diff
@@ -48,21 +59,53 @@ var ERMrest = (function(module) {
     /**
      * @function
      * @param {Object} element a model element (schema, table, or column)
-     * @return {String}
+     * @param {Object} parentElement the upper element (schema->null, table->schema, column->table)
      * @desc This function determines the display name for the schema, table, or
      * column elements of a model.
      */
-    module._determineDisplayName = function (element) {
+    module._determineDisplayName = function (element, parentElement) {
         var displayname = element.name;
+        var hasDisplayName = false;
         try {
             var display_annotation = element.annotations.get("tag:misd.isi.edu,2015:display");
-            if (display_annotation && display_annotation.content &&
-                display_annotation.content.name) {
-                displayname = display_annotation.content.name;
+            if (display_annotation && display_annotation.content) {
+
+                //get the specified display name
+                if(display_annotation.content.name){
+                    displayname = display_annotation.content.name;
+                    hasDisplayName = true;
+                }
+
+                //get the name styles
+                if(display_annotation.content.name_style){
+                    element._nameStyle = display_annotation.content.name_style;
+                }
             }
         } catch (exception) {
             // no display annotation, don't do anything
         }
+
+        // if name styles are undefined, get them from the parent element
+        // Note: underline_space and title_case might be null.
+        if(parentElement){
```

Comment:
right, it doesn't have to literally be recursive. so yes as long as you are hitting up the whole chain thats fine.